app/scripts/controllers/buildConfig.js

```diff
@@ -110,6 +114,28 @@ angular.module('openshiftConsole')
           $scope.imageSourcesPaths.push($filter('destinationSourcePair')(imageSource.paths));
         });
       }
+      var buildFrom = _.get(buildStrategy(buildConfig), 'from', {});
+      // We don't want to reload the image every time the BC updates, only load again if the from changes
+      var buildFromImageKey = buildFrom.kind + "/" + buildFrom.name + "/" + (buildFrom.namespace || $scope.projectName);
+      if (lastLoadedBuildFromImageKey !== buildFromImageKey) {
+        if (_.includes(["ImageStreamTag", "ImageStreamImage"], buildFrom.kind)) {
+          lastLoadedBuildFromImageKey = buildFromImageKey;
+          DataService.get(APIService.kindToResource(buildFrom.kind), buildFrom.name, {namespace: buildFrom.namespace || $scope.projectName}).then(function(imageStreamImage){
+            $scope.BCEnvVarsFromImage = _.map(_.get(imageStreamImage, 'image.dockerImageMetadata.Config.Env'),
```

Comment:
Should be able to use `$filter('imageEnv')`

https://github.com/spadgett/origin-web-console/blob/ca4be237df6935819e795eeb2962e985230876e1/app/scripts/filters/resources.js#L287-L298